File: src/DataScience/R/bridge.R
make_device <- function(name) {
  list(
    power_on = function() paste0(name, ":on"),
    mute = function() paste0(name, ":muted")
  )
}

basic_remote <- function(device) {
  list(activate = function() device$power_on())
}

mute_remote <- function(device) {
  list(activate = function() device$mute())
}

tv <- make_device("TV")
radio <- make_device("Radio")

cat("basic-tv=", basic_remote(tv)$activate(), "\n", sep = "")
cat("basic-radio=", basic_remote(radio)$activate(), "\n", sep = "")
cat("mute-tv=", mute_remote(tv)$activate(), "\n", sep = "")
cat("mute-radio=", mute_remote(radio)$activate(), "\n", sep = "")

File: src/Other/MicroPython/bridge.py
class Device:
    def __init__(self, name):
        self.name = name

    def power_on(self):
        return self.name + ":on"

    def mute(self):
        return self.name + ":muted"


class BasicRemote:
    def __init__(self, device):
        self.device = device

    def activate(self):
        return self.device.power_on()


class MuteRemote:
    def __init__(self, device):
        self.device = device

    def activate(self):
        return self.device.mute()


tv = Device("TV")
radio = Device("Radio")
print("basic-tv=" + BasicRemote(tv).activate())
print("basic-radio=" + BasicRemote(radio).activate())
print("mute-tv=" + MuteRemote(tv).activate())
print("mute-radio=" + MuteRemote(radio).activate())
